src/views/blockDetails/blockBody/index.tsx

```diff
@@ -0,0 +1,236 @@
+                  { addSuffix: true },
+                );
+                return (
+                  <tr
+                    key={extrinsic.id}
+                    className={cn(
+                      'pd-table-row',
+                      ' text-dev-purple-600 dark:text-dev-purple-550',
+                    )}
+                    onClick={() => handleOpenModal(extrinsic)}
+                  >
+                    <td>{extrinsic.id}</td>
+                    <td>{blockNumber}</td>
+                    <td>{timeAgo}</td>
+                    <td>
+                      {extrinsic.isSuccess
+                        ? (
+                          <Icon
+                            size={[16]}
+                            name="icon-checked"
+                            className="text-dev-green-600"
+                          />
+                        )
+                        : (
+                          <Icon
+                            size={[16]}
+                            name="icon-clock"
+                            className="animate-rotate text-dev-yellow-700"
+                          />
+                        )
+                      }
+                    </td>
+                    <td>{extrinsic.method.section} ({extrinsic.method.method})</td>
+                    <td>
+                      <Icon
+                        size={[18]}
+                        name="icon-dropdownArrow"
+                        className="text-dev-black-1000 dark:text-dev-purple-50"
+                      />
+                    </td>
+                  </tr>
+                );
+              })}
+            </tbody>
+          </table>
+        </div>
+      )}
+      {filter === 'events' && (
+        <div className="overflow-x-auto">
+          <table className="w-full">
+            <colgroup>
+              <col className="min-w-28" />
+              <col className="min-w-36" />
+              <col className="min-w-36" />
+              <col className="min-w-10" />
+            </colgroup>
+            <thead>
+              <tr className={cn(
+                'bg-dev-purple-100 text-left',
+                'dark:bg-dev-black-900',
+              )}
+              >
+                <th className="py-4 pl-2 font-geist font-body2-bold">Event ID</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold">Action</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold">Type</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold" />
+              </tr>
+            </thead>
+            <tbody>
+              {visibleEvents.map((event: IMappedBlockEvent, eventIndex: number) => (
+                <tr
+                  key={eventIndex}
+                  className={cn(
+                    'pd-table-row',
+                    ' text-dev-purple-600 dark:text-dev-purple-550',
+                  )}
+                  onClick={() => handleOpenModal(event)}
```

Comment:
no! ... we already have visibleEvents ... we need only the index here and useCallback